queries.sql

```diff
@@ -8,4 +8,31 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = TRUE;
 SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+SELECT * FROM animals;
+COMMIT;
+SELECT * FROM animals;
+BEGIN;
```

Comment:
Hey @Al-Amin-Khan-Shakil,

First and foremost, great job on your project so far! Your dedication and effort shine through, and I truly appreciate the work you've done.

However, I did notice one area in your submission that we need to revisit, specifically regarding the transaction task with the `animals` table.

**Required Transaction Rollback Query**:
The project explicitly stated:

> "Inside a transaction delete all records in the animals table, then roll back the transaction."

To align with the project's requirements, your SQL commands should look something like this:

```sql
BEGIN;

DELETE FROM animals;   --  Delete ALL records
SELECT COUNT(*) FROM animals;   -- This will confirm the records were deleted

ROLLBACK;
SELECT COUNT(*) FROM animals;   -- This will show the records are back after the rollback
```

Please adjust your code to this structure and resubmit. If you have any questions or need clarification, I'm here to help.

Once again, fantastic job so far! With this adjustment, you'll be right on track.

Warm regards,

Temitope